scripts/release_notes.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import subprocess
+import json
+import os
+import re
+import sys
+
+if len(sys.argv) < 2:
+    #      0         1         2         3         4         5         6         7         8
+    #      012345678901234567890123456789012345678901234567890123456789012345678901234567890
+    print("Usage: ./release_notes.py <ancestor-sha>", file=sys.stderr)
+    print()
+    print("Extract release notes from git commits from <ancestor-sha> (exclusive) to HEAD", file=sys.stderr)
+    print("(inclusive).", file=sys.stderr)
+    sys.exit(1)
+
+
+def git(*args):
+    return subprocess.check_output(["git"] + list(args)).decode()
+
+
+RE_HEADING = re.compile(
+    r"#+ Release notes(.*)",
+    re.DOTALL | re.IGNORECASE,
+)
+
+RE_CHECK = re.compile(
+    r"^\s*-\s*\[.\]",
```

Comment:
Is there a case where someone could format as [ x ]? `^\s*-\s*\[\s?.\s?\]` maybe? Can't get too loose just in case the note involves a list of markdown links.